ledger/src/block_location_lookup.rs

```diff
@@ -0,0 +1,36 @@
+//! Lookup from a repair nonce to the location where this shred should be stored
+//! When requesting repair we insert the (Nonce, Location) pair here.
+//! Once the shred passes sigverify checks, the Location is fetched before insertion into blockstore.
+//!
+//! This lookup avoids having to pass the location information around, or polluting
+//! the OutstandingRequests cache with this extra information and lock contention.
+//!
+//! We don't actively cleanup this cache, its size is managed according to Lru policy
+
+use {
+    crate::{blockstore_meta::BlockLocation, shred::Nonce},
+    lru::LruCache,
+    std::sync::{Arc, RwLock},
+};
+
+pub struct BlockLocationLookup {
```

Comment:
what was the reason we couldn't augment the existing `OutstandingRequests` cache with the location?

Reply:
outstanding requests verification is performed before sigverify and we pop the request as soon as we've received the appropriate amount of responses. So this would require holding onto the request for longer and only popping it off once we've additionally sigverified the response.
Seemed cleaner to split this out into a different lookup.